integration-cli/docker_cli_help_test.go

```diff
@@ -279,87 +245,59 @@ func (s *DockerSuite) TestHelpExitCodesHelpOutput(c *check.C) {
-	stdout, stderr, ec, err = runCommandWithStdoutStderr(cmd)
-	if len(stdout) == 0 || len(stderr) != 0 || ec != 0 || err != nil {
-		c.Fatalf("Bad results from 'docker --help'\nec:%d\nstdout:%s\nstderr:%s\nerr:%v", ec, stdout, stderr, err)
-	}
+	out, _, err = dockerCmdWithError("--help")
+	c.Assert(err, checker.IsNil, check.Commentf(out))
 	// Be really pick
-	if strings.HasSuffix(stdout, "\n\n") {
-		c.Fatalf("Should not have a blank line at the end of 'docker --help'\n%s", stdout)
-	}
+	c.Assert(out, checker.Not(checker.HasSuffix), "\n\n", check.Commentf("Should not have a blank line at the end of 'docker --help'\n"))
 
 	// docker inspect busybox: stdout=all, stderr=empty, rc=0
 	// Just making sure stderr is empty on valid cmd
-	cmd = exec.Command(dockerBinary, "inspect", "busybox")
-	stdout, stderr, ec, err = runCommandWithStdoutStderr(cmd)
-	if len(stdout) == 0 || len(stderr) != 0 || ec != 0 || err != nil {
-		c.Fatalf("Bad results from 'docker inspect busybox'\nec:%d\nstdout:%s\nstderr:%s\nerr:%v", ec, stdout, stderr, err)
-	}
+	out, _, err = dockerCmdWithError("inspect", "busybox")
+	c.Assert(err, checker.IsNil, check.Commentf(out))
 	// Be really pick
-	if strings.HasSuffix(stdout, "\n\n") {
-		c.Fatalf("Should not have a blank line at the end of 'docker inspect busyBox'\n%s", stdout)
-	}
+	c.Assert(out, checker.Not(checker.HasSuffix), "\n\n", check.Commentf("Should not have a blank line at the end of 'docker inspect busyBox'\n"))
 
 	// docker rm: stdout=empty, stderr=all, rc!=0
 	// testing the min arg error msg
-	cmd = exec.Command(dockerBinary, "rm")
-	stdout, stderr, ec, err = runCommandWithStdoutStderr(cmd)
-	if len(stdout) != 0 || len(stderr) == 0 || ec == 0 || err == nil {
-		c.Fatalf("Bad results from 'docker rm'\nec:%d\nstdout:%s\nstderr:%s\nerr:%v", ec, stdout, stderr, err)
-	}
+	cmd := exec.Command(dockerBinary, "rm")
+	stdout, stderr, _, err := runCommandWithStdoutStderr(cmd)
+	c.Assert(err, checker.NotNil)
+	c.Assert(len(stderr), checker.Not(checker.Equals), 0)
+	c.Assert(stdout, checker.Equals, "")
 	// Should not contain full help text but should contain info about
 	// # of args and Usage line
-	if !strings.Contains(stderr, "requires a minimum") {
-		c.Fatalf("Missing # of args text from 'docker rm'\nstderr:%s", stderr)
-	}
+	c.Assert(stderr, checker.Contains, "requires a minimum", check.Commentf("Missing # of args text from 'docker rm'\n"))
 
 	// docker rm NoSuchContainer: stdout=empty, stderr=all, rc=0
 	// testing to make sure no blank line on error
 	cmd = exec.Command(dockerBinary, "rm", "NoSuchContainer")
-	stdout, stderr, ec, err = runCommandWithStdoutStderr(cmd)
-	if len(stdout) != 0 || len(stderr) == 0 || ec == 0 || err == nil {
-		c.Fatalf("Bad results from 'docker rm NoSuchContainer'\nec:%d\nstdout:%s\nstderr:%s\nerr:%v", ec, stdout, stderr, err)
-	}
+	stdout, stderr, _, err = runCommandWithStdoutStderr(cmd)
+	c.Assert(err, checker.NotNil)
+	c.Assert(len(stderr), checker.Not(checker.Equals), 0)
```

Comment:
This one I think it should be added as the content check below is checker.Not(checker.HasSuffix), "" can be passed